app/controllers/archives_controller.rb

```diff
@@ -1,8 +1,18 @@
 class ArchivesController < ApplicationController
   def index
-    @archives = {}
-    current_user.bank_accounts.each do |account|
-      @archives[account] = account.archives.order(date: :desc)
+    @year = params[:year].to_i == 0 ? Date.today.year : params[:year].to_i
+    @account = params[:account].present? ? current_user.bank_accounts.find(params[:account]) : current_user.bank_accounts.first
```

Comment:
This should be extracted in provate methods `find_year` and `find_account` that are called in a `before_action`